docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,804 @@
+}
+```
+
+###### Public Key
+
+*Note:* Here we decoupled the `Address` type from its corresponding `PubKey`. The corresponding codec step is proposed to be abstracted out from the CryptoProvider layer.
+
+```go
+type PubKey interface {
+	Bytes() []byte
+	Equals(other PubKey) bool
+	Type() string
+}
+```
+
+###### Private Key
+
+*Note*: For example, in hardware wallets, the `PrivKey` interface acts only as a *reference* to the real data. This is a design consideration and may be subject to change during implementation.
+
+Future enhancements could include additional security functions such as **zeroing** memory after private key usage to further enhance security measures.
+
+
+```go
+type PrivKey interface {
+	Bytes() []byte
+	PubKey() PubKey
+	Equals(other PrivKey) bool
+	Type() string
+}
+```
+
+##### Storage and persistence
+
+The storage and persistence layer is tasked with storing a `CryptoProvider`s. Specifically, this layer must:
+
+* Securely store the crypto provider's associated private key (only if stored locally, otherwise a reference to the private key will be stored instead).
+* Store the `ProviderMetadata` struct which contains the data that distinguishes that provider.
+
+The purpose of this layer is to ensure that upon retrieval of the persisted data, we can access the provider's type, version, and specific configuration (which varies based on the provider type). This information will subsequently be utilized to initialize the appropriate factory, as detailed in the following section on the factory pattern.
+
+The storage proposal involves using a modified version of the [Record](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/crypto/keyring/v1/record.proto) struct, which is already defined in **Keyring/v1**. Additionally, we propose utilizing the existing keyring backends (keychain, filesystem, memory, etc.) to store these `Record`s in the same manner as the current **Keyring/v1**.
+
+*Note: This approach will facilitate a smoother migration path from the current Keyring/v1 to the proposed architecture.*
+
+Below is the proposed protobuf message to be included in the modified `Record.proto` file
+
+###### Protobuf message structure
+
+```protobuf
+
+// cryptoprovider.proto
+
+syntax = "proto3";
+
+package crypto;
+
+import "google/protobuf/any.proto";
+
+// CryptoProvider holds all necessary information to instantiate and configure a CryptoProvider.
+message CryptoProvider {
+    string name = 1; // (unique) name of the crypto provider.
+    google.protobuf.Any pub_key = 2;
+    string type = 3;       // Type of the crypto provider
+    string version = 4;    // Version (semver format)
+    map<string, bytes> config = 5;  // Configuration data with byte array values
+    google.protobuf.Any privKey = 6; // Optional if key is stored locally
+}
+```
+<b>name</b>:
+Specifies the unique name of the crypto provider. This name is used to identify and reference the specific crypto provider instance.
```

Comment:
Can you provide a couple of examples here (as you do below with `type`)

Reply:
The idea is that `name` can be any string the user chooses, so together with `version` and `type` we have a proper identifier. For example: `ledger001.LedgerCryptoProvider.v1.0.0` (`<name>.<type>.<version>`)